Add service image support with dark theme variants and translate the services

Mohamed-Elhanafy opened the pull request:
## Summary
This PR introduces service image support throughout the application with dedicated dark theme variants. It replaces the previous icon-based system with dynamic image loading for services, enhancing the visual experience and maintaining consistency across light and dark themes.

## Changes
- Added `imageUrl` and `darkImageUrl` fields to Service and ServiceDto entities
- Replaced static icons with dynamic AsyncImage components in UI
- Updated RequestBottomSheetContent to support dark theme service images
- Modified CategoryItem and ServiceTypeCard components to handle image loading with dark theme support
- Added dark theme detection logic to properly switch between light and dark service images
- Updated view models to fetch and manage service image data
- Removed deprecated icon resource mapping utilities

## Key Features
- Services now display actual images instead of static icons
- Automatic switching between light and dark theme images based on system/app theme
- Improved UI consistency across different sections (home, requests, account)

<img width="460" height="895" alt="image" src="https://github.com/user-attachments/assets/ef926754-c831-4019-b5ad-4bb7d8232c90" />

<img width="518" height="901" alt="image" src="https://github.com/user-attachments/assets/a710c14a-0e13-43d0-8268-3227a14aaebc" />

melhanafy-otas committed:
Integrate service image URL and update UI components

melhanafy-otas committed:
add support for dark theme service images in various components

Mohamed-Elhanafy committed:
Merge branch 'develop' into refactor/service-images

Joseph-Sameh-0 commented on app/src/main/java/com/paris_2/san3a/presentation/screen/account/AccountViewModel.kt:
```diff
@@ -757,7 +757,9 @@ class AccountViewModel(
-                        description = mapOf(currentLocale to serviceUiState.serviceDescription)
+                        description = mapOf(currentLocale to serviceUiState.serviceDescription),
+                        imageUrl = serviceUiState.serviceImage,
+                        darkImageUrl = serviceUiState.serviceImage
```
```suggestion
                        darkImageUrl = serviceUiState.serviceDarkImage
```

Joseph-Sameh-0 commented on app/src/main/java/com/paris_2/san3a/presentation/screen/home/customer/CustomerHomeViewModel.kt:
```diff
@@ -37,29 +37,49 @@ class CustomerHomeViewModel(
+    private fun findServiceImageUrl(serviceId: String): String {
+        val services = screenState.value.customerUiState.services
+        val mostRequested = screenState.value.customerUiState.mostRequestedServices
+        return (services + mostRequested).find { it.id == serviceId }?.imageUrl ?: ""
```
I think it's better if you but a default image or fake one

Joseph-Sameh-0 commented on app/src/main/java/com/paris_2/san3a/presentation/screen/requests/component/MyJobOfferCard.kt:
```diff
@@ -54,7 +54,9 @@ fun MyJobOfferCard(
                     serviceType = jobUiState.serviceType,
-                    serviceId = jobUiState.serviceId,
+                    serviceImageUrl = jobUiState.serviceImage,
+                    darkServiceImageUrl = jobUiState.serviceImage,
```
same image ?

Joseph-Sameh-0 commented on app/src/main/java/com/paris_2/san3a/presentation/screen/requests/component/RequestCard.kt:
```diff
@@ -67,7 +67,13 @@ fun RequestCard(
+                title = requestUi.requestTitle,
+                serviceType = requestUi.serviceType,
+                serviceImageUrl = requestUi.serviceImage,
+                darkServiceImageUrl = requestUi.serviceImage,
```
and here

Joseph-Sameh-0 committed:
Merge branch 'develop' into refactor/service-images

Joseph-Sameh-0 committed:
Refactor: Add serviceImage parameter to JobUiState conversion

Joseph-Sameh-0 committed:
Refactor: Remove darkImageUrl and handle theme in mappers

- Remove `darkImageUrl` from `Service` entity and related UI components.
- Update `ServiceMapper` to select `imageUrl` or `darkImageUrl` based on the current theme.
- Modify `HomeRepositoryImpl` and `UserRepositoryImpl` to use `LocalDataStore` to get theme status and pass it to the mapper.
- Adjust `CustomerHomeScreen`, `RequestBottomSheetContent`, `CategoryItem`, `MostRequestedSection`, `MyServiceViewModel`, and `AccountViewModel` to reflect the removal of `darkImageUrl` and rely on the mapper for theme-specific image URLs.

Joseph-Sameh-0 committed:
Refactor: Remove dark theme logic from customer home screen

This commit removes the dark theme specific logic from the customer home screen and related components.

- Removed `CustomizeProfileSettingsUseCase` dependency and `loadAppTheme` function from `CustomerHomeViewModel`.
- Removed `isDarkTheme` property from `CustomerHomeUiState`.
- Removed `isDarkTheme` parameter from `CustomerHomeScreen`'s `MostRequestedSection` composable.
- Removed `isDarkTheme` and `darkServiceImageUrl` parameters and related logic from `CategoryItem`, `CategoryItemLarge`, and `CategoryItemSmall` composables. The image displayed will now solely depend on `serviceImageUrl`.
- Removed `isDarkTheme` parameter from `MostRequestedSection` composable.

Joseph-Sameh-0 committed:
Refactor: Remove dark theme image logic from ServiceTypeCard

This commit simplifies the `ServiceTypeCard` component by removing the logic for displaying different images in dark theme.

- The `darkServiceImageUrl` and `isDarkTheme` parameters have been removed from `ServiceTypeCard`.
- The `ServiceTypeCard` will now always display the image specified by `serviceImageUrl`.
- Usages of `ServiceTypeCard` in `RequestCard` and `MyJobOfferCard` have been updated to reflect this change.

Joseph-Sameh-0 committed:
Merge branch 'develop' into refactor/service-images

Joseph-Sameh-0 committed:
refactor: remove unused dark theme check from CategoryItem

Joseph-Sameh-0 committed:
Merge branch 'develop' into refactor/service-images

Joseph-Sameh-0 committed:
Refactor: Update bottom sheet initialization

- Pass `imageUrl` directly to `initBottomSheet` function in `CustomerHomeInteractionListener`, `CustomerHomeViewModel`, and `CustomerHomeScreen`.
- Remove `findServiceImageUrl` function from `CustomerHomeViewModel` as it's no longer needed.
- Update `MostRequestedSection` to pass `imageUrl` when a service is clicked.

Joseph-Sameh-0 committed:
Refactor: Update service data handling and localization

- Modify `Service` entity to store `title` and `description` as single strings instead of maps.
- Update `ServiceMapper` to select the appropriate language for `title` and `description` based on the current app language, defaulting to English if the specific language is not available.
- Change `saveServices` in `UserRepository`, `UserRemoteDataSource`, `SetUpAccountUseCase`, `AccountViewModel`, and `MyServiceViewModel` to accept a list of service IDs (strings) instead of a list of `Service` objects.
- Adjust `CustomerHomeViewModel` and `CustomerHomeScreen` to use the new `Service` entity structure and pass the `Service` object directly to `onServiceClick`.
- Remove language-specific constants (`ARABIC_NAME`, `ENGLISH_NAME`, etc.) from `CustomerHomeScreen`.
- Update `HomeRepositoryImpl` to pass the current app language to the `toEntity` mapper function when fetching and searching services.
- Simplify `mapServiceToUiState` by directly using the `title` and `description` from the `Service` entity.
- Update `MostRequestedServices` composable to accept a `Service` object in its click listener.

Joseph-Sameh-0 committed:
Refactor: Remove unused RequestDetailsStatus enum and CustomerRequestDetails data class

- Delete `RequestDetailsStatus` enum from `RequestService.kt` as it is no longer used.
- Remove the unused `CustomerRequestDetails` data class from `CustomerRequestDetailsScreenState.kt`.
- Add a TODO comment in `CustomerHomeScreen.kt` to use the app language for voice recognition.

Joseph-Sameh-0 committed:
Refactor: Mark `getSuggestions` for removal

The `getSuggestions` function in `SuggestionMapper.kt` has been marked with a TODO comment for future removal.

Joseph-Sameh-0 committed:
Refactor: Streamline Customer Home BottomSheet and Request Logic

This commit refactors the customer home screen's bottom sheet and request creation process.

- Modified `CustomerHomeInteractionListener` and `CustomerHomeViewModel` to pass the entire `Service` object to `initBottomSheet` instead of individual properties like `serviceTitle`, `serviceId`, and `imageUrl`.
- Updated `CustomerHomeUiState` and `BottomSheetUiState` to store the `Service` object directly, removing redundant fields.
- Adjusted `CustomerHomeScreen` to utilize the `Service` object from `BottomSheetUiState` for displaying service details in the bottom sheet steps.
- Ensured the "Next" button in the `SELECT_LOCATION` step of the bottom sheet is only active when address details, government ID, and city ID are all provided.
- Removed unused constants (`ARABIC_NAME`, `ENGLISH_NAME`, etc.) from `CustomerHomeViewModel`.
- Removed the `color` and `containImg` parameters from `RequestBottomSheetContent` composable as they were either unused or had fixed values. The title color is now consistently `Theme.colors.additional.primary.blue`.
- Added a `TODO` comment in `CustomerHomeUiState` regarding handling localized error messages.
- Added a `TODO` comment in `CustomerHomeViewModel` to potentially replace `serviceType` (title) with `serviceId` in `RequestServiceUiState`.

Joseph-Sameh-0 committed:
Enhance Craftsman UI and add localizations

- Add new string resources for "Jobs Done", "Earnings", "Rating", and "EGP" currency format, and their Arabic translations.
- Update `RequestCardForCraftsMan` to:
    - Accept `offers` as `Int` instead of `String`.
    - Use `stringResource` for "number_of_offers".
    - Trim description text and limit it to one line with ellipsis overflow.
    - Minor padding and icon adjustments.
- Update `CraftsmanHomeScreen` to pass `offersCount` as `Int` to `RequestCardForCraftsMan`.
- Update `StatsContainer` to:
    - Use `stringResource` for "Jobs Done", "Earnings", and "Rating".
    - Display `jobsDone`, `earnings` (with currency format), and `rating` as strings.
- Remove redundant error handling for null offers in `MyJobsCraftsmanViewModel`.

Joseph-Sameh-0 committed:
Refactor: Improve craftsman stats display and update Arabic translations

- Refactor `StatsContainer` in craftsman home screen to use a reusable `StatsColumn` composable for displaying individual stats (jobs done, earnings, rating).
- Update Arabic translations for "jobs_done" to "الوظائف المكتملة" and "egp" to "ج.م".

Joseph-Sameh-0 committed:
Refactor: Integrate service suggestions from backend

- Add `suggestions` field to `ServiceDto` and `Service` entities to hold service-specific suggestions.
- Update `ServiceMapper` to include `suggestions` mapping based on language and theme.
- Modify `ServiceDto.fromJson` to parse `suggestions` from the remote data source.
- Update `CustomerHomeScreen` to use `suggestions` from `bottomSheetService` instead of the local `getSuggestions` function.
- Delete unused `SuggestionMapper.kt` file.
- Remove unused import in `StatsContainer.kt`.

Joseph-Sameh-0 committed:
Refactor: Improve job display and data handling

- Update `CustomerHomeViewModel` to remove unnecessary variables in service filtering.
- Modify `RequestCardForCraftsMan` to use `AsyncImage` for displaying service images and adjust padding.
- Update `MyRequestCustomerViewModel` to ensure `servicesList` is not null before processing and adjust function call order.
- Refactor `CraftsmanHomeScreen` to pass `imageUri` to `RequestCardForCraftsMan`.
- In `CraftsmanHomeViewModel`:
    - Change `userServices` in `CraftsmanHomeUiState` from `List<String>` to `Map<String, Service>`.
    - Fetch user services before loading available jobs and recent related jobs.
    - Use service image URL when mapping jobs to `RequestServiceUiState`.
- Update `RequestServiceUiState` to include `imageUrl` and pass it during mapping in `CraftsmanHomeUiState.kt`.

Joseph-Sameh-0 committed:
Merge branch 'develop' into refactor/service-images
